### ISSUE TITLE
Update new(ish) skeleton sprites

### ISSUE THREAD

**Issue by doofus-01** [wesnoth/wesnoth#8150]
### Describe the desired feature

Death squire and skeleton riders got merged to core after some discussion I'd missed at the time (#6088)
 - The riders don't have very skeleton-like resistances
 - Should the squire have an axe instead of sword?  I didn't see that get resolved
 - The sprites should be updated

It's probably too late to do anything about the resistances, but the other two should still be on the table.
I don't have this far along enough for a PR yet, but this is a first draft (still with a sword)
![Screenshot_20231225_skele-demo](https://github.com/wesnoth/wesnoth/assets/13510196/b679121f-d272-4712-a8d9-707955233b58)

**Comment by sevu** [comment]
> It's probably too late to do anything about the resistances

Just yesterday a similar change was made, so probably fine

**Comment by Hejnewar** [comment]
No that change was for deep water movement and submerge. 

That said, Im fine with their current resistances and they can stay as a mix between horse and skeleton. It could be better but is fine as is.

**Comment by doofus-01** [comment]
While working on these, another question came to me:  **What are the bobbing skeleton animations supposed to represent?** 

I had issues with the heaving animations of most units as default standing animations because I think the sprites are supposed to convey information (as well as flavor and entertainment), and if I saw someone heaving like that in real life, I'd figure we need to get that guy out of the field or else he's going to collapse.  But a skeleton?  Is it a bunch of floating objects that sync to look like bobbing?  Maybe...

A "wounded" skeleton wouldn't have any reason to heave or be woozy, but maybe the "puppet strings" are damaged or tangled, so it has an unstable & uncoordinated bobbing animation.  That's what I went with for the skeleton rider in #8156, but it would be nice to get some sort of convention on the skeleton standing animations.

(And BTW, these skeleton units were merged into mainline without meeting some minimal standard, this shouldn't be how it is done.  I wasn't paying enough attention this time, but I will revert something like 4878c35 in the future, if it comes up again and I still have that privilege.)
